File: java-recipes/src/main/java/com/fargopolis/enums/AbilitySource.java
package com.fargopolis.enums;

import java.util.Map;

public enum AbilitySource {
    CLASS(1),
    RACE(2),
    FEAT(3),
    OTHER(4);

    private Integer value;

    private static final Map<Integer, AbilitySource> BY_VALUE = Map.of(
            1, CLASS,
            2, RACE,
            3, FEAT,
            4, OTHER);

    AbilitySource(Integer value) {
        this.value = value;
    }

    public Integer getValue() {
        return value;
    }

    public static AbilitySource getByValue(Integer value) {
        return BY_VALUE.get(value);
    }
}

File: recipe-site/app/_constants/AbilitySource.ts
export enum AbilitySource {
    Class = "CLASS",
    Race = "RACE",
    Feat = "FEAT",
    Other = "OTHER"
}

export const ABILITY_SOURCE_OPTIONS = [
    { value: AbilitySource.Class, label: "Class" },
    { value: AbilitySource.Race, label: "Race" },
    { value: AbilitySource.Feat, label: "Feat" },
    { value: AbilitySource.Other, label: "Other" }
];

export function getLabelForAbilitySource(source: AbilitySource): string {
    return ABILITY_SOURCE_OPTIONS.find(o => o.value === source)?.label ?? "Unknown";
}